packages/design-system/src/scalars/components/url-field/url-favicon.tsx

```diff
@@ -0,0 +1,47 @@
+import { IconName, iconNames } from "@/powerhouse";
+import { Icon } from "@/powerhouse/components/icon";
+// import { type PlatformIcon } from "./url-field";
+import { useMemo } from "react";
+
+interface UrlFaviconProps {
+  url: string;
+  platformIcons?: Record<string, React.ReactElement | IconName>;
+}
+
+const DEFAULT_ICON_NAME = "GlobeWww";
+
+const UrlFavicon = ({ url, platformIcons }: UrlFaviconProps) => {
+  const IconToUse: React.ReactElement | null = useMemo(() => {
+    const defaultIcon = <Icon name={DEFAULT_ICON_NAME} size={18} />;
+    try {
+      const parsedUrl = new URL(url);
+      const hostname = parsedUrl.hostname;
+      const icon = platformIcons?.[hostname];
+      if (!icon) return defaultIcon;
+
+      if (typeof icon === "string") {
+        // check beyond typescript check if the icon actually exists
+        if (iconNames.includes(icon)) {
+          return <Icon name={icon} size={18} />;
+        } else {
+          console.error(`Platform icon "${icon}" does not exist`);
+          return defaultIcon;
+        }
+      }
+
+      return icon;
+    } catch {
+      return defaultIcon;
+    }
+  }, [url, platformIcons]);
+
+  if (Object.keys(platformIcons ?? {}).length === 0) return null;
```

Comment:
I think if we move this declaration
if (Object.keys(platformIcons ?? {}).length === 0) return null;
 up of the useMemo, we can avoid some calculations in the case that the conditions match